pkg/monitor/serialization/resource.go

```diff
@@ -0,0 +1,75 @@
+package monitorserialization
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kube-openapi/pkg/util/sets"
+)
+
+func InstanceMapToFile(filename string, resourceType string, instances monitorapi.InstanceMap) error {
+	namespaceToKeys := map[string]sets.String{}
+	for key, obj := range instances {
+		ns := "---missing|metadata---"
+		metadata, err := meta.Accessor(obj)
+		if err == nil {
+			ns = metadata.GetNamespace()
+		}
+
+		if _, ok := namespaceToKeys[ns]; !ok {
+			namespaceToKeys[ns] = sets.String{}
+		}
+		namespaceToKeys[ns].Insert(key)
+	}
+
+	nsToItems := map[string]*unstructured.UnstructuredList{}
+	for _, ns := range sets.StringKeySet(namespaceToKeys).List() {
+		nsList := &unstructured.UnstructuredList{}
+		for _, instanceKey := range namespaceToKeys[ns].List() {
+			instanceMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(instances[instanceKey])
+			if err != nil {
+				return err
+			}
+			nsList.Items = append(
+				nsList.Items,
+				unstructured.Unstructured{
+					Object: instanceMap,
+				},
+			)
+		}
+		nsToItems[ns] = nsList
+	}
+
+	byteBuffer := &bytes.Buffer{}
+	zipWriter := zip.NewWriter(byteBuffer)
+
+	for namespace, nsItems := range nsToItems {
+		json, err := json.MarshalIndent(nsItems, "", "  ")
+		if err != nil {
+			return err
+		}
+		nsWriter, err := zipWriter.Create(filepath.Join(namespace, resourceType+".json"))
```

Comment:
Can we put these all in a top level directory to avoid dumping their contents in whatever unfortunate directory is $PWD?

Reply:
> Can we put these all in a top level directory to avoid dumping their contents in whatever unfortunate directory is $PWD?

sure, will update later.